Refactor go

ErlendHaa opened the pull request.

ajaust reviewed (commented):
The changes look good to me. However, we now have the word `query` left at quite a few places as part of directories, file names and within code (see below). Are we sure that the word `request` is now used at all relevant places?

## File and paths with keyword "query"

```text
$ find . -iname "*query*"
./cmd/query
./api/query_test.gotext
```

```text
$ rg -i query
api/request.go
42:// Query for slice endpoints
43:// @Description Query payload for slice endpoint /slice.
                                                
api/endpoint.go
104:    if err := json.Unmarshal([]byte(ctx.Query("query")), v); err != nil {
118:// @Param    query  query  string  True  "Urlencoded/escaped MetadataRequest"
155:// @Param    query  query  string  True  "Urlencoded/escaped SliceRequest"
174:// @Param    body  body  SliceRequest  True  "Query Parameters"
194:// @Param    query  query  string  True  "Urlencoded/escaped FenceResponse"
                                                
api/query_test.go                                                                               
28:     slice          testSliceQuery
29:     sliceQuery     string                                                                   
37:     fence          testFenceQuery                                                           
38:     fenceQuery     string                                                                   
46:     metadata       testMetadataQuery
47:     metadataQuery  string                                                                   
57:type testSliceQuery struct {                                                                 
64:type testFenceQuery struct {       
71:type testMetadataQuery struct {                                                              
93:                     name:   "Valid GET Query",                         
95:                     slice: testSliceQuery{                                                  
104:                    name:   "Valid json POST Query",                       
106:                    slice: testSliceQuery{
209:                    name:           "Invalid json GET query",
211:                    sliceQuery:     "help I am a duck",                
216:                    name:           "Invalid json POST query",
218:                    sliceQuery:     "help I am a duck",
223:                    name:   "Missing parameters GET query",
225:                    sliceQuery: "{\"vds\":\"" + well_known +
231:                    name:   "Missing parameters POST Query",              
233:                    sliceQuery: "{\"vds\":\"" + well_known +     
239:                    name:   "Query with unknown axis",
241:                    slice: testSliceQuery{
251:                    name:   "Query which passed all input checks but still should fail",
253:                    slice: testSliceQuery{
289:                    name:   "Valid GET Query",
291:                    fence: testFenceQuery{
300:                    name:   "Valid json POST Query",               
302:                    fence: testFenceQuery{
367:                    name:           "Invalid json GET query",
369:                    fenceQuery:     "help I am a duck", 
374:                    name:           "Invalid json POST query",
376:                    fenceQuery:     "help I am a duck",
381:                    name:   "Missing parameters GET query",
383:                    fenceQuery: "{\"vds\":\"" + well_known +
389:                    name:   "Missing parameters POST Query",
391:                    fenceQuery: "{\"vds\":\"" + well_known +
397:                    name:   "Query with unknown coordinate system",
399:                    fence: testFenceQuery{
409:                    name:   "Query which passed all input checks but still should fail",
411:                    fence: testFenceQuery{
447:                    name:   "Valid GET Query",
449:                    metadata: testMetadataQuery{
456:                    name:   "Valid json POST Query",
458:                    metadata: testMetadataQuery{
510:                    name:           "Invalid json GET query",
512:                    metadataQuery:  "help I am a duck",
517:                    name:           "Invalid json POST query",
519:                    metadataQuery:  "help I am a duck",
524:                    name:           "Missing parameters GET query",
526:                    metadataQuery:  "{\"vds\":\"" + well_known + "\"}",
531:                    name:           "Missing parameters POST Query",
533:                    metadataQuery:  "{\"sas\":\"somevalidsas\"}",
538:                    name:   "Query which passed all input checks but still should fail",
540:                    metadata: testMetadataQuery{
637:    method string, query interface{}, jsonQuery string) {
638:    jsonRequest := jsonQuery
639:    if jsonQuery == "" {
640:            req, err := json.Marshal(query)
656:            q.Add("query", jsonRequest)
657:            ctx.Request.URL.RawQuery = q.Encode()
672:    prepareRequest(ctx, t, "/slice", testcase.method, testcase.slice, testcase.sliceQuery)
676:    prepareRequest(ctx, t, "/fence", testcase.method, testcase.fence, testcase.fenceQuery)
680:    prepareRequest(ctx, t, "/metadata", testcase.method, testcase.metadata, testcase.metadat
aQuery)

docs/fence.md
23:       querying /metadata

tests/e2e/test_requests.py
112:@pytest.mark.parametrize("path, query", [
128:def test_assure_no_unauthorized_access(path, query, sas, error_msg):
129:    query.update({"sas": sas})
131:                       params={"query": json.dumps(query)})
137:@pytest.mark.parametrize("path, query", [
142:def test_assure_only_allowed_storage_accounts(path, query):
143:    query.update({
147:                       params={"query": json.dumps(query)})
153:@pytest.mark.parametrize("path, query, error_code, error", [
191:def test_errors(path, query, error_code, error):
194:    query.update({"sas": sas})
195:    res = requests.post(f'{ENDPOINT}/{path}', json=query)
206:    query = make_slice_request(VDSURL, direction, lineno, sas)
207:    json_query = json.dumps(query)
208:    encoded_query = urllib.parse.quote(json_query)
211:        rdata = requests.get(f'{ENDPOINT}/slice?query={encoded_query}')
214:        rdata = requests.post(f'{ENDPOINT}/slice', json=query)
237:    query = make_fence_request(VDSURL, coordinate_system, coordinates, sas)
238:    json_query = json.dumps(query)
239:    encoded_query = urllib.parse.quote(json_query)
242:        rdata = requests.get(f'{ENDPOINT}/fence?query={encoded_query}')
245:        rdata = requests.post(f'{ENDPOINT}/fence', json=query)
263:    query = make_metadata_request(VDSURL, sas)
264:    json_query = json.dumps(query)
265:    encoded_query = urllib.parse.quote(json_query)
268:        rdata = requests.get(f'{ENDPOINT}/metadata?query={encoded_query}')
271:        rdata = requests.post(f'{ENDPOINT}/metadata', json=query)

tests/performance/helpers/report-helpers.js
4:  "checks{status checks:query}": [{ threshold: "rate == 1.00" }],

tests/performance/helpers/request-helpers.js
55:  const queryResStatusCheck = check(
58:      "query response: status must be 200": (r) => r.status === 200,
60:    { "status checks": "query" }
62:  if (!queryResStatusCheck) {
63:    fail(`Wrong 'query' response status: ${res.status}`);

Dockerfile
45:RUN /tools/swag init -g cmd/query/main.go --md docs
93:ENTRYPOINT [ "/server/query" ]
```

ErlendHaa commented:
Good catch! I've renamed query_test.go to endpoint_test.go and substituted query for response in that file. I've also updated the other files where it made sense

ajaust reviewed (approved):
Looks good to me. I also like the change from `query` to payload in the tests.
